example/remove/t8_test_elements_remove.cxx

```diff
@@ -0,0 +1,261 @@
+{
+  int                 level, min_level, max_level;
+  t8_cmesh_t          cmesh;
+  t8_forest_t         forest, forest_1, forest_2;
+  t8_scheme_cxx_t    *scheme;
+
+  struct t8_adapt_data adapt_data = {{{1.0, 0.5, 0.5},
+                                      {0.5, 1.0, 0.5},
+                                      {0.5, 0.5, 1.0},
+                                      {0.0, 0.5, 0.5},
+                                      {0.5, 0.0, 0.5},
+                                      {0.5, 0.5, 0.0}}};
+
+  scheme = t8_scheme_new_default_cxx ();
+  /* Construct a cmesh */
+  cmesh = t8_test_create_cmesh (cmesh_id);
+  /* Compute the first level, such that no process is empty */
+  min_level = t8_forest_min_nonempty_level (cmesh, scheme);
+  /* Use one level with empty processes */
+  min_level = SC_MAX (min_level, 3);
+  max_level = min_level + 1;
+  
+  for (level = min_level; level < max_level; level++) {
+    t8_cmesh_ref (cmesh);
+    forest = t8_forest_new_uniform (cmesh, scheme, level, 0, sc_MPI_COMM_WORLD);
+
+    forest_1 = t8_forest_new_adapt (forest  , t8_adapt_callback_init  , 0, 0, &adapt_data);
+    forest_1 = t8_forest_new_adapt (forest_1, t8_adapt_callback_remove, 0, 0, &adapt_data);
+
+    t8_forest_ref (forest_1);
+    forest_2 = t8_forest_new_adapt (forest_1, t8_adapt_callback_coarse, 0, 0, &adapt_data);
+    forest_2 = t8_forest_new_adapt (forest_2, t8_adapt_callback_refine, 0, 0, &adapt_data);
+    forest_2 = t8_forest_new_adapt (forest_2, t8_adapt_callback_remove, 0, 0, &adapt_data);
+
+    SC_CHECK_ABORT (t8_forest_is_equal(forest_1, forest_2),
+                    "The forests are not equal");
+
+    t8_scheme_cxx_ref (scheme);
+    t8_forest_unref (&forest_1);
+    t8_forest_unref (&forest_2);
+  }
+  t8_scheme_cxx_unref (&scheme);
+  t8_cmesh_destroy (&cmesh);
+}
+
+void
+test_cmesh_emelemts_remove_all ()
+{
+  /* Test all cmeshes over all different inputs we get through their id */
+  for (int cmesh_id = 0; cmesh_id < t8_get_number_of_all_testcases ();
+       cmesh_id++) {
+    /* This if statement is necessary to make the test work by avoiding specific cmeshes which do not work yet for this test.
+     * When the issues are gone, remove the if statement. */
+    if (cmesh_id != 6 && cmesh_id != 89 && (cmesh_id < 237 || cmesh_id > 256)) {
+      /* Skip all t8_test_create_new_bigmesh_cmesh 
+       * ToDo: Add geometrie to bigmeshs
+       */
+      if (cmesh_id < 97 || cmesh_id > 256) {
+        t8_test_emelemts_remove(cmesh_id);
+      }
+    }
+  }
+}
+
+int
+main (int argc, char **argv)
+{
+  int                 mpiret;
+  sc_MPI_Comm         mpic;
+
+  mpiret = sc_MPI_Init (&argc, &argv);
+  SC_CHECK_MPI (mpiret);
+
+  mpic = sc_MPI_COMM_WORLD;
+  sc_init (mpic, 1, 1, NULL, SC_LP_PRODUCTION);
+  p4est_init (NULL, SC_LP_ESSENTIAL);
+  t8_init (SC_LP_DEFAULT);
+
+  test_cmesh_emelemts_remove_all ();
+
+  sc_finalize ();
+
+  mpiret = sc_MPI_Finalize ();
+  SC_CHECK_MPI (mpiret);
+
+  return 0;
+}
```

Comment:
This is a very good test for your new features! Please move it into the test-files. That way they are run when compiling with make check.